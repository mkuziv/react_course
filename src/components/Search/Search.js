import React from 'react';
import SearchButton from './SearchButton/SearchButton';
import SearchInput from './SearchInput/SearchInput';

import './Search.css';

const Search = ({ searchQuery, setSearchQuery, setIsSubmitted }) => {
  function handleSubmit(e) {
    e.preventDefault();
    setIsSubmitted(true);
  }
  return (
    <form className="search-form" onSubmit={handleSubmit}>
      <SearchInput searchQuery={searchQuery} setSearchQuery={setSearchQuery} />
      <SearchButton />
    </form>
  )
};

export default Search;