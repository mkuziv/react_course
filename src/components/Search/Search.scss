@import '../../styles/variables.scss';
@import '../../styles/mixin.scss';

.search {
  margin-bottom: 10px;
  background: url('../../images/header_bg.png');  
  background-repeat: no-repeat;
  background-size: cover;
  height: 396px;
  padding-top: 100px;

  .wrapper {
    max-width: 960px;
    margin: 0 auto;
    display: flex;
    flex-direction: column;

    .h2 {
      margin: 35px 0;
      font-style: normal;
      @include text(40px, 300, 49px);
      letter-spacing: 1px;
      text-transform: uppercase;  
      color: $white;
    }
  
    .search-form {
      display: flex;
      justify-content: space-between;

      .btn-big {
        padding: 16px 74px;
      }
    }
  }
}


