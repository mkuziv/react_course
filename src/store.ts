import { configureStore } from '@reduxjs/toolkit';
import { useDispatch } from 'react-redux';
import postsSlice from './slice/postsSlice';

const store = configureStore({
  reducer: {
    posts: postsSlice,
  },
});
export const useAppDispatch = () => useDispatch<typeof store.dispatch>();
export default store;
