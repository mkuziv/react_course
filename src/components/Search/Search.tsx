import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import SearchInput from './SearchInput/SearchInput';
import Button from '../Button/Button';
import { fetchPosts, selectQuery, updateSearch } from '../../slice/postsSlice';
import { useAppDispatch } from '../../store';
import getQueryParams from '../../utils/getQueryParams';

import './Search.scss';

const Search = () => {
  const [searchQuery, setSearchQuery] = useState('');

  const dispatch = useAppDispatch();
  const query = useSelector(selectQuery);

  const handleSubmit = async (e: React.FormEvent) => {
    e.preventDefault();
    await dispatch(updateSearch(searchQuery));
    const searchQueryValue = getQueryParams(query);

    console.warn('searchQueryValue', searchQueryValue);
    await dispatch(fetchPosts(searchQueryValue));
  };

  return (
    <section className="search">
      <div className="wrapper">
        <h2 className="h2">FIND YOUR MOVIE</h2>
        <br />
        <form className="search-form" onSubmit={handleSubmit}>
          <SearchInput searchQuery={searchQuery} setSearchQuery={setSearchQuery} />
          <Button type="submit" content="search" btn="btn btn-big btn-red" />
        </form>
      </div>
    </section>
  );
};

export default Search;
