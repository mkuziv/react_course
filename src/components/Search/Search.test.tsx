import React from 'react';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../store';
import Search from './Search';

const renderWithProviders = (children: JSX.Element) => (
  render(
    <MemoryRouter>
      <Provider store={store}>
        {children}
      </Provider>
    </MemoryRouter>,
  )
);

describe('Search', () => {
  it('render snapshot', () => {
    const { container } = renderWithProviders(<Search />);
    expect(container).toMatchSnapshot();
  });

  it('check input value', () => {
    renderWithProviders(<Search />);
    const input = screen.getByPlaceholderText('What do you want to watch?');
    expect((input as HTMLInputElement).value).toBe('');
    fireEvent.change(input, { target: { value: 'Cool' } });

    screen.debug();
    expect((input as HTMLInputElement).value).toBe('Cool');
  });

  it('check click', () => {
    renderWithProviders(<Search />);
    const input = screen.getByPlaceholderText('What do you want to watch?');
    const button = screen.getByText('search');
    fireEvent.change(input, { target: { value: 'Cool' } });

    fireEvent.click(button);
    expect((input as HTMLInputElement).value).toBe('Cool');
  });
});
