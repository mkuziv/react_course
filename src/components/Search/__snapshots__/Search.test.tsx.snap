// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search render snapshot 1`] = `
<div>
  <section
    class="search"
  >
    <div
      class="wrapper"
    >
      <h2
        class="h2"
      >
        FIND YOUR MOVIE
      </h2>
      <br />
      <form
        class="search-form"
      >
        <input
          class="search-input"
          placeholder="What do you want to watch?"
          type="text"
          value=""
        />
        <button
          class="btn btn-big btn-red"
          type="submit"
        >
          search
        </button>
      </form>
    </div>
  </section>
</div>
`;
